scripts/btrfs-balance

```diff
@@ -73,83 +69,68 @@ report_progress()
 	echo "BALANCE_PROGRESS:$1"
 }
 
-# Keep device out of suspend while we balance
-keepalive $SUSPEND_LIMIT&
-PID_KEEPALIVE=$!
-
-keepalive_cleanup()
-{
-	kill $PID_KEEPALIVE
-	exit $1
-}
-
-trap -- 'keepalive_cleanup 1' TERM HUP SEGV INT ILL FPE ABRT
-
 while getopts "d:t:c:fh" opt; do
 	case $opt in
 	d)
-		if ! test -e $OPTARG; then
+		if ! test -e "$OPTARG"; then
 			echo "Root device \"$OPTARG\" does not exist!" >&2
-			keepalive_cleanup 1
+			exit 1
 		fi
-		ROOTDEV=$OPTARG
+		ROOTDEV="$OPTARG"
 	;;
 	t)
-		if test $OPTARG -ge 0 && test $OPTARG -le 100; then
+		if test "$OPTARG" -ge 0 && test "$OPTARG" -le 100; then
 			BALANCE_THRESHOLD=$OPTARG
 		else
 			echo "Error: -t $OPTARG is out of 0-100 range!"
-			keepalive_cleanup 1
+			exit 1
 		fi
 	;;
 	c)
-		if test $OPTARG -ge 0 && test $OPTARG -le 100; then
+		if test "$OPTARG" -ge 0 && test "$OPTARG" -le 100; then
 			CHARGE_THRESHOLD=$OPTARG
 		else
 			echo "Error: -c $OPTARG is out of 0-100 range!"
-			keepalive_cleanup 1
+			exit 1
 		fi
 	;;
 	f)
 		get_usages
-		echo $FREE_SPACE
-		keepalive_cleanup 0
+		echo "$FREE_SPACE"
+		exit 0
 	;;
 	h)
 		print_help
-		keepalive_cleanup 0
+		exit 0
 	;;
 	\?)
 		echo "Invalid option: -$OPTARG" >&2
 		print_help
-		keepalive_cleanup 1
+		exit 1
 	;;
 	:)
 		print_help
-		keepalive_cleanup 1
+		exit 1
 	;;
 
 	esac
 done
 
 check_balance_need
-RETVAL=$?
-if test $RETVAL -eq 1; then
-	keepalive_cleanup 1
-fi
+NEED_BALANCE=$?
 
-if test $NEED_BALANCE == 1; then
+if test "$NEED_BALANCE" == 1; then
```

Comment:
Now that `NEED_BALANCE` is not used anywhere else (as far as I can see), this:

```
check_balance_need
NEED_BALANCE=$?
if test "$NEED_BALANCE" == 1; then
```

Can be collapsed into:

```
if check_balance_need; then
```